eds_scikit/io/arrow.py

```diff
@@ -0,0 +1,57 @@
+import os
+
+import pandas as pd
+import pyarrow as pa
+
+
+class arrowConnector:  # pragma: no cover
+    def __init__(self, sql, database_name: str):
+        self.database_name = database_name
+        self.db = (
+            sql(f"DESCRIBE DATABASE EXTENDED {database_name}")
+            .filter("database_description_item=='Location'")
+            .collect()[0]
+            .database_description_value
+        )
+
+    def get_pd_fragment(
+        self,
+        table_name: str,
+    ):
+
+        self.path_table = os.path.join(self.db, table_name)
+
+        # Import the parquet as ParquetDataset
+        parquet_ds = pa.parquet.ParquetDataset(
+            self.path_table, use_legacy_dataset=False
+        )
+
+        # Partitions of ds
+        fragments = iter(parquet_ds.pieces)
+
+        yield from fragments
```

Comment:
As far as I understand, `pa.parquet.ParquetDataset` loads metadata about parquet files as a list, so what is the benefit of `yield from` over `return` here?

Reply:
TBH it was crashing when returning the full list, I wasn't able to fully grasp what was happening, but using a generator did the trick !